backend/transaction/abstract_use_case.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABC, abstractmethod
+
+
+class AbstractTransactionUseCase(ABC):
+    """
+    Abstract base class defining the contract for transaction use cases.
+    """
+
+    @abstractmethod
+    def upload_data_to_firestore_use_case(self):
+        """
+        Abstract method for uploading data to Firestore.
+        Implementations should handle writing transaction data to Firestore.
+        """
+        pass
```

Comment:
Perhaps use "raise NotImplementedError" here for abstract methods.

Reply:
Good catch!